includes/class-wc-payments-apple-pay-registration.php

```diff
@@ -434,4 +483,76 @@ public function display_error_notice() {
+		if ( 'yes' !== $this->gateway->get_option( 'payment_request' ) ) {
+			return;
+		}
+
+		if ( $this->can_merchant_register_domain_with_applepay( $this->account->get_account_country() ) ) {
+			return;
+		}
+
+		if ( ! $this->is_account_in_supported_applepay_countries( $this->account->get_account_country() ) &&
+			'1' !== get_user_meta( get_current_user_id(), 'dismissed_applepay_not_in_supported_countries_notice', true ) ) {
+			?>
+			<div id="wcpay-applepay-error" class="notice notice-error woocommerce-message">
+				<a class="woocommerce-message-close notice-dismiss" href="<?php echo esc_url( wp_nonce_url( add_query_arg( 'wc-hide-notice', 'applepay_not_in_supported_countries' ), 'woocommerce_hide_notices_nonce', '_wc_notice_nonce' ) ); ?>"><?php esc_html_e( 'Dismiss', 'woocommerce-payments' ); ?></a>
+				<p>
+					<b><?php esc_html_e( 'Apple Pay: ', 'woocommerce-payments' ); ?></b>
+					<?php
+					echo sprintf(
+						/* translators: 1: supported country list */
+						__( 'Apple Pay is not currently supported in your country. <a href="%1$s">Countries and regions that support Apple Pay (Apple Support)</a>', 'woocommerce-payments' ), // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+						'https://support.apple.com/en-us/HT207957'
+					);
+					?>
+				</p>
+			</div>
+			<?php
+		}
+
+		if ( ! $this->has_domain_association_file_permissions() ) {
+			/**
+			 * In case domain association file is missing or the DOMAIN_ASSOCIATION_FILE_DIR folder is inaccessible,
+			 * permalink structure should be checked to evaluate if error or warning message should be displayed.
+			 */
+			if ( '' === get_option( 'permalink_structure', '' ) ) {
+				?>
+				<div id="wcpay-applepay-error" class="notice notice-error">
+					<p>
+						<b><?php esc_html_e( 'Apple Pay: ', 'woocommerce-payments' ); ?></b>
+						<?php
+							echo sprintf(
+								/* translators: 1: apple pay support */
+								__( 'We were not able to verify your domain. <a href="%1$s" target="_blank">This help documentation</a> will walk you through the process to verify with Apple that you control your domain.', 'woocommerce-payments' ), // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+								'https://woocommerce.com/document/payments/apple-pay/#apple-pay-button-does-not-appear'
+							);
+						?>
+					</p>
+				</div>
+				<?php
+			} else {
+				?>
+				<div id="wcpay-applepay-warning" class="notice notice-warning">
+					<p>
+						<b><?php esc_html_e( 'Apple Pay: ', 'woocommerce-payments' ); ?></b>
+						<?php
+							echo sprintf(
+								/* translators: 1: apple pay support */
+								__( 'We were not able to access your domain verification file. WooCommerce Payments will keep on working, but we recommend using <a href="%1$s" target="_blank">this documentation</a> to walk you through the process to verify with Apple that you control your domain.', 'woocommerce-payments' ), // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
This string feels weird. Could a native English speaker take a look?

Reply:
Updated in e7382294083d0c9f1dd4f3ada333e00d34024a02